### DISCUSSION TITLE
Dependencies UC ESS & Knowledge Agents (Kit)

### DISCUSSION THREAD

**Discussion by drcgjung** [eclipse-tractusx/sig-release#200] [Ideas]
This is the list of features needed for shipping an Environmental & Social Standards use case based on the Agents Kit for a release 24.03

- Simple Graph Exchanger
  - Architecture Revision: Separate Matchmaking Agent from EDC Agent Plane
- Simple Graph Exchanger Frontend
  - Extend the SDE for accessing the Matchmaking Agent or
  - Have a new "SGE" application
- Abstract Supply Chain Ontology
  - Trace Skill should be able to deal with BPNL, BPNS & BPNA
  - Trace Skill should be able to deal with "arbitrary" hierarchical taxonomies.

**Comment by drcgjung** [comment]
# Feature "Refactor Agent Plane-Matchmaking Agent Interface for Enabling Simple Graph Exchanger"

## Benefit Hypothesis
* the Simple Graph Exchanger Feature has already been developed in a preview branch of KA-EDC
* It adds the ability to host business data from the RDF Store over the Matchmaking Agent
* The Matchmaking Agent is currently integrated into the Agent Plane container that is bound to be public
* After the refactoring the Matchmaking Agent and RDF Store are in a separate container and Agent Plane and Matchmaking Agent communicate over a (private) network interface

## Beneficiaries
* small and medium sized companies which want to directly upload/provide use case data
* security-aware companies who want to separate data storage from public API even for meta-data.

## Acceptance Criteria
* Interface definition for Matchmaking Agent
* Matchmaking Agent + Datspace Synchronizer + RDF Store Code transferred to KA-RI
* Deployment of Matchmaking Agent
* Configuration of Matchmaking Endpoint in KA-EDC
* Interfacing the new Interface from KA-EDC

Estimated Story Points:
30 SP

**Comment by drcgjung** [comment]
# Feature "Simple Graph Exchanger Application"

## Benefit Hypothesis
* Using the SGE Application, it will be easy to manually upload data supporting particular KA-based use cases
* It will be possible to change and/or delete data
* It will be possible to control under which policies and contracts the data is published
* It will be possible to limit with which dataspace/internet resources the data can be merged/integrated

## Beneficiaries
* small and medium sized companies which want to directly upload/provide use case data

## Acceptance Criteria
* Either a separate application or a functionality integrated into the SDE
* Should be publishable/compatible with the Catena-X Portal
* Upload of data/creation of assets should be possible given use case templates
* Changing of data and deletion of data should be possible.

Estimated Story Points:
30 SP

**Comment by drcgjung** [comment]
# Feature "Supply-Chain Domain Ontology"

## Benefit Hypothesis
* using always digital twin/serial parts information is a too strong requirement for many use cases 
* furthermore, the existing "hard-typed" relationships can be flexiblizized level (legal<->site<->address)
* many use cases can be based on purely abstract supply-chain information.
* the supply chain domain ontology will be able to relate the business partners based on a plethora of concrete data sources and diverse taxonomical conventioins.

## Beneficiaries
* any use case/use case participant based on supply chain information

## Acceptance Criteria
* domain ontology created
* domain ontology valid
* sample data bindings, use case templates
* Demonstration of integration of various level of data

Estimated Story Points:
13 SP

**Comment by stephanbcbauer** [comment]
@drcgjung Since this idea is one year old and i assume the features are already implemented, I close the discussion.
